.github/workflows/update_submodules.yaml

```diff
@@ -0,0 +1,29 @@
+name: Submodule Updater
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "submodules"
+
+jobs:
+  Submodules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 1
+          persist-credentials: false
+          submodules: recursive
+      - name: Checkout submodules
+        uses: textbook/git-checkout-submodule-action@master
```

Comment:
github caches docker images we use in the workflow, it wont pull mast from him again unless we force it to. It can be pinned to a commit or version though.